haystack/translator/transformers.py

```diff
@@ -0,0 +1,139 @@
+import logging
+from typing import Any, Dict, List, Optional, Union
+
+from transformers import AutoModelForSeq2SeqLM, AutoTokenizer
+
+from haystack import Document
+from haystack.translator.base import BaseTranslator
+
+logger = logging.getLogger(__name__)
+
+
+class TransformersTranslator(BaseTranslator):
+    """
+    Translator component based on Seq2Seq models from Huggingface's transformers library.
+    Exemplary use cases:
+    - Translate a query from Language A to B (e.g. if you only have good models + documents in language B)
+    - Translate a document from Language A to B (e.g. if you want to return results in the native language of the user)
+
+    We currently recommend using OPUS models (see __init__() for details)
+
+    **Example:**
+
+        ```python
+        |    DOCS = [
+        |        Document(text="Heinz von Foerster was an Austrian American scientist combining physics and philosophy,
+        |                       and widely attributed as the originator of Second-order cybernetics.")
+        |    ]
+        |    translator = TransformersTranslator(model_name_or_path="Helsinki-NLP/opus-mt-en-de")
+        |    res = translator.translate(documents=DOCS, query=None)
+        ```
+    """
+    def __init__(
+        self,
+        model_name_or_path: str,
+        tokenizer_name: Optional[str] = None,
+        max_length: Optional[int] = 512,
```

Comment:
I am fine with adding this param. However, we should name it `max_seq_len` to be consistent with other parts in Haystack (e.g. Reader). I think it's also better to use `None` as the default so that when calling `AutoTokenizer.from_pretrained()`, we use the original value from the respective tokenizer config.